### PR TITLE
feat(ec2): add m5zn instances

### PR DESCRIPTION
**Pull request by robertd**
`m5zn` instances release notes:
https://aws.amazon.com/blogs/aws/new-ec2-m5zn-instances-fastest-intel-xeon-scalable-cpu-in-the-cloud/

Docs have already been updated a while ago:
https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-instance.html#cfn-ec2-instance-instancetype

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by robertd** [comment]
@njlynch These are based on m5, so let me know if `HIGH_COMPUTE_MEMORY2` is a proper name to use here. Release notes mention that `M5zn instances feature high frequency processing, are a variant of the general-purpose M5 instances, and are built on the AWS Nitro System.`... therefore they kinda are high compute & memory. Thoughts?

File: packages/@aws-cdk/aws-ec2/lib/instance-types.ts [hunk 1]
```diff
@@ -566,6 +566,16 @@ export enum InstanceClass {
    */
   Z1D = 'z1d',
 
+  /**
+   * High memory and compute capacity instances based on Intel Xeon Scalable (Cascade Lake) processors, 2nd generation
+   */
+  HIGH_COMPUTE_MEMORY2 = 'm5zn',
```
Comment (njlynch) [inline packages/@aws-cdk/aws-ec2/lib/instance-types.ts:572]: Hmm... 

I get that the blog post mentions this as building on the success of the z1d instances, but this does belong to the "m" family, so I think we should reflect that in the name (e.g., `STANDARD5_HIGH_COMPUTE`?) and description.

**Review by njlynch — CHANGES_REQUESTED** [review]

File: packages/@aws-cdk/aws-ec2/lib/instance-types.ts (hunk 1 above)
Comment (robertd) [reply packages/@aws-cdk/aws-ec2/lib/instance-types.ts]: Updated!

Comment (robertd) [reply packages/@aws-cdk/aws-ec2/lib/instance-types.ts]: Please review and lmk. TY!

**Review by njlynch — APPROVED** [review]

**Pull request merged**